src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -572,6 +574,74 @@ private static bool HasApplicableConditionalMethod(OverloadResolutionResult<Meth
             return result;
         }
 
+        private BoundExpression BindLocalFunctionInvocationWithDynamicArgument(
+            CSharpSyntaxNode syntax,
+            CSharpSyntaxNode expression,
+            string methodName,
+            BoundMethodGroup methodGroup,
+            DiagnosticBag diagnostics,
+            CSharpSyntaxNode queryClause,
+            MethodGroupResolution resolution)
+        {
+            // Invocations of local functions with dynamic arguments
+            // don't need to be dispatched as dynamic invocations
+            // since they cannot be overloaded.  Instead, we'll just
+            // emit a standard call with dynamic implicit conversions
+            // for any dynamic arguments. The one exception is params
+            // arguments which cannot be targeted by dynamic arguments
+            // because there is an ambiguity between an array target
+            // and the params element target. See
+            // https://github.com/dotnet/roslyn/issues/10708
+            Debug.Assert(resolution.OverloadResolutionResult.Succeeded);
```

Comment:
I was wondering how you handle "call by name" part when you do not know the mangled name of the method. 

I see how :-)  It is not supported for now.